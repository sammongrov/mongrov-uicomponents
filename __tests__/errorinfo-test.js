import React from 'react';
import renderer from 'react-test-renderer';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { Alert } from 'react-native';
import Errorinfo from '../ErrorInfo';

configure({ adapter: new Adapter() });
const error = () => Alert.alert('message');

it('Text renders correctly', () => {
  const tree = renderer.create(<Errorinfo />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('Errorinfo Component using shallow renders correctly', () => {
  const errorType = 'error';
  const title = 'message';
  const wrapper = shallow(<Errorinfo title={title} errorType={errorType} error={error} />);
  expect(wrapper.instance().props.error).toMatchSnapshot();
});

test('Errorinfo Component using shallow renders correctly', () => {
  const errorType = 'warning';
  const title = 'message';
  const wrapper = shallow(<Errorinfo title={title} errorType={errorType} error={error} />);
  expect(wrapper.instance().props.error).toMatchSnapshot();
});
