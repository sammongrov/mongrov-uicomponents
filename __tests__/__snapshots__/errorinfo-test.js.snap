// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Errorinfo Component using shallow renders correctly 1`] = `[Function]`;

exports[`Errorinfo Component using shallow renders correctly 2`] = `[Function]`;

exports[`Text renders correctly 1`] = `<View />`;
